pkg/relayer/session/session.go

```diff
@@ -81,6 +82,15 @@ type relayerSessionsManager struct {
 
 	// bankQueryClient is used to query for the bank module parameters.
 	bankQueryClient client.BankQueryClient
+
+	// stopping is a boolean that indicates whether the relayerSessionsManager is
+	// in the process of a graceful shutting down.
+	// During normal operation, context cancellations (e.g. deadlines) are treated
+	// as failures that should cause failing session trees to be deleted.
+	// When stopping=true, these same cancellations are expected as part of shutdown
+	// and should not trigger deletion, ensuring session trees are properly persisted
+	// for recovery after restart.
+	stopping bool
```

Comment:
I kind of feel like this should be an `atomic.bool`

```bash
import "sync/atomic"

stopping atomic.Bool
```

The concurrency scares me given that it's "just a bool"

Reply:
It is under lock, but atomic bool is a good practice.